File: Leetcode/hashmap/500_KeyboardRow/Solution.java
class Solution {
    public String[] findWords(String[] words) {
    /*/ One liner
    return Stream.of(words).filter(s -> s.toLowerCase().matches("[qwertyuiop]*|[asdfghjkl]*|[zxcvbnm]*")).toArray(String[]::new);
    /*/
        String row1 = "qwertyuiop";
        String row2 = "asdfghjkl";
        String row3 = "zxcvbnm";
        HashMap<Character, Integer> map = new HashMap<>();
        for(int i = 0; i<row1.length(); i++){
            map.put(row1.charAt(i), 1);
        }
        for(int i = 0; i<row2.length(); i++){
            map.put(row2.charAt(i), 2);
        }
        for(int i = 0; i<row3.length(); i++){
            map.put(row3.charAt(i), 3);
        }
        List<String> word = new ArrayList<>();
        for(String s: words){
            boolean same = true;
            int row = map.get(Character.toLowerCase(s.charAt(0)));
            for(int i = 1; i<s.length(); i++){
                if( row != map.get(Character.toLowerCase(s.charAt(i))) ){
                    same = false;
                    break;
                }
            }
            if(same)
                word.add(s);
        }
        String[] res = new String[word.size()];
        word.toArray(res);
        return res;
    }
}

File: Leetcode/hashmap/500_KeyboardRow/Solution.py
class Solution(object):
    def findWords(self, words):
        """
        :type words: List[str]
        :rtype: List[str]
        """
        """
        row1 = set("qwertyuiopQWERTYUIOP")
        row2 = set("asdfghjklASDFGHJKL")
        row3 = set("ZXCVBNMzxcvbnm")
        return filter(lambda x: set(x).issubset(row1) or set(x).issubset(row2) or set(x).issubset(row3), words)
        """
        row1 = "qwertyuiop"
        row2 = "asdfghjkl"
        row3 = "zxcvbnm"
        res = []
        for s in words:
            c1 = sum(map(s.lower().count, row1))
            c2 = sum(map(s.lower().count, row2))
            c3 = sum(map(s.lower().count, row3))
            if (c1 and c2) + (c2 and c3) +(c1 and c3) == 0:
                res.append(s)
        return res
